RecoHGCal/TICL/plugins/PFTICLProducer.cc

```diff
@@ -90,13 +92,15 @@ void PFTICLProducer::produce(edm::Event& evt, const edm::EventSetup& es) {
     const auto abs_pdg_id = std::abs(ticl_cand.pdgId());
     const auto charge = ticl_cand.charge();
     const auto& four_mom = ticl_cand.p4();
-    double ecal_energy = 0.;
-
+    double ecal_energy_fraction = 0.;
     for (const auto& t : ticl_cand.tracksters()) {
```

Comment:
it seems you are only evaluating ecal_energy_fraction for the last trackster in the for loop.   In that case you wouldn't need a for loop, you could simply access the element at the end.  I suspect that's not really what you want to do though.

Reply:
The code seems logically wrong. You cannot consider only the last component. And, at the same time, you cannot sum up the fractions and apply the sum to the overall energy.
Now, while this could be the only available approximation for the case in which the _regressed energy_ is used, for the `raw_energy` case the exact algebra has to be restored.